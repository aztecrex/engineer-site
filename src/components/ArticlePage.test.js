import React from 'react';
const createMockStore = configureMockStore([]);
import ezJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';

import ConnectedArticlePage, {ArticlePage,mapStateToProps} from './ArticlePage';

describe('unconnected', () => {

  it('renders happy', () => {
    // given
    const id = 'article309';
    const title = 'This is SO  Neat';

    // when
    const rendered = shallow(<ArticlePage {...{id,title}} />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });

  it('shows not found when no id', () =>{

    // when
    const rendered = shallow(<ArticlePage />);

    // then
    expect(ezJson(rendered)).toMatchSnapshot();


  });
});

describe('map state', () => {

  it('maps params id and title', () => {
    // given
    const id = 'brave';
    const title = 'Brave Little Engine';
    const state = {
      articles: {
        index: {
          [id]: {
            title: title
          }
        }
      }
    };

    // when
    const actual = mapStateToProps(state, {params: {id:id}});

    // then
    expect(actual).toEqual({id,title});
  });

  it('missing index entry results in null id', () => {
    // given
    const id = 'brave';
    const state = {
      articles: {
        index: {
          notBrave: {
            title: 'Oh Yeah?'
          }
        }
      }
    };

    // when
    const actual = mapStateToProps(state, {params: {id:id}});

    // then
    expect(actual).toEqual({id:null});

  });

  it('missing index entry results in null id', () => {
    // given
    const state = {
      articles: {
        index: {
          anId: {
            title: 'Does not really matter'
          }
        }
      }
    };

    // when
    const actual = mapStateToProps(state, {params: {}});

    // then
    expect(actual).toEqual({});

  });


});

describe('connected', () => {

  it('is connected to state', () => {
    // given
    const id = 'article4';
    const title = 'Title of Article 4';
    const state = {
      articles: {
        index: {
          [id]: {title}
        }
      }
    };
    const params = {id};
    const store = createMockStore(state);

    // when
    const rendered = shallow(
      <ConnectedArticlePage store={store} params={params} /> );

    // then
    expect(ezJson(rendered)).toMatchSnapshot();

  });

});
