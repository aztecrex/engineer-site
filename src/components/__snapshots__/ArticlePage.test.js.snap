exports[`connected is connected to state 1`] = `
<ArticlePage
  dispatch={[Function]}
  id="article4"
  params={
    Object {
      "id": "article4",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "notify": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": null,
    }
  }
  title="Title of Article 4" />
`;

exports[`unconnected renders happy 1`] = `
<div
  className="page home">
  <Flexbox
    display="flex"
    element="div"
    flexDirection="row">
    <Flexbox
      display="flex"
      element="div"
      flexDirection="column">
      <Flexbox
        display="flex"
        element="header">
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/">
          <span
            id="title">
            Engineer Site
          </span>
        </Link>
      </Flexbox>
      <hr />
      <Flexbox
        display="flex"
        element="div"
        flexGrow={1}>
        <Connect(Article)
          id="article309" />
      </Flexbox>
      <hr />
      <Flexbox
        display="flex"
        element="footer">
        <Connect(Harper) />
        <hr />
        <div>
          <p>
            Revision 
            <Revision />
          </p>
        </div>
      </Flexbox>
    </Flexbox>
    <Flexbox
      display="flex"
      element="div"
      flex="0 0 20em">
      <Connect(Directory) />
    </Flexbox>
  </Flexbox>
</div>
`;

exports[`unconnected shows not found when no id 1`] = `
<div
  className="not-found-page">
  Not Found
</div>
`;
